core/client/fs/src/main/java/alluxio/client/block/stream/DefaultBlockWorkerClient.java

```diff
@@ -0,0 +1,141 @@
+
+  private static final long KEEPALIVE_TIMEOUT_MS =
+      Configuration.getMs(PropertyKey.USER_NETWORK_GRPC_KEEPALIVE_TIMEOUT_MS);
+  private static final long GRPC_FLOWCONTROL_WINDOW =
+      Configuration.getMs(PropertyKey.USER_NETWORK_GRPC_FLOWCONTROL_WINDOW);
+  private static final long MAX_INBOUND_MESSAGE_SIZE =
+      Configuration.getMs(PropertyKey.USER_NETWORK_GRPC_MAX_INBOUND_MESSAGE_SIZE);
+  private static final EventLoopGroup WORKER_GROUP = NettyUtils
+      .createEventLoop(NettyUtils.USER_CHANNEL_TYPE,
+          Configuration.getInt(PropertyKey.USER_NETWORK_NETTY_WORKER_THREADS),
+          "netty-client-worker-%d", true);
+
+  private GrpcChannel mChannel;
+  private BlockWorkerGrpc.BlockWorkerBlockingStub mBlockingStub;
+  private BlockWorkerGrpc.BlockWorkerStub mAsyncStub;
+
+  /**
+   * Builder for {@link DefaultBlockWorkerClient}.
+   */
+  public static class Builder implements BlockWorkerClient.Builder {
+    private final GrpcChannelBuilder mChannelBuilder;
+
+    /**
+     * Creates a {@link Builder} for {@link DefaultBlockWorkerClient}.
+     * @param channelBuilder a gRPC channel builder
+     */
+    public Builder(GrpcChannelBuilder channelBuilder) {
+      mChannelBuilder = channelBuilder;
+    }
+
+    @Override
+    public BlockWorkerClient build() {
+      return new DefaultBlockWorkerClient(mChannelBuilder.build());
+    }
+  }
+
+  /**
+   * Gets a builder for given user subject and address.
+   *
+   * @param subject the user subject, can be null if the user is not available
+   * @param address the address of the worker
+   * @return the builder for the client
+   */
+  public static Builder getBuilder(Subject subject, SocketAddress address) {
+    return new Builder(GrpcChannelBuilder
+        .forAddress(address)
+        .channelType(NettyUtils.getClientChannelClass(
+            !(address instanceof InetSocketAddress)))
+        .group(WORKER_GROUP)
+        .usePlaintext(true)
+        .keepAliveTimeout(KEEPALIVE_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+        .maxInboundMessageSize((int) MAX_INBOUND_MESSAGE_SIZE)
+        .flowControlWindow((int) GRPC_FLOWCONTROL_WINDOW));
+  }
+
+  /**
+   * Creates a client instance for communicating with block worker.
+   *
+   * @param channel the gRPC channel
+   */
+  public DefaultBlockWorkerClient(GrpcChannel channel) {
+    mChannel = channel;
+    mBlockingStub = BlockWorkerGrpc.newBlockingStub(mChannel);
+    mAsyncStub = BlockWorkerGrpc.newStub(mChannel);
+  }
+
+  @Override
+  public boolean isShutdown() {
+    ConnectivityState state = mChannel.getState(false);
+    return state == ConnectivityState.SHUTDOWN;
+  }
+
+  @Override
+  public void close() throws IOException {
+    mChannel.shutdown();
+    try {
+      if (!mChannel.awaitTermination(
+          Configuration.getMs(PropertyKey.WORKER_NETWORK_NETTY_SHUTDOWN_TIMEOUT),
+          TimeUnit.MILLISECONDS)) {
+        LOGGER.warn("Failed to close gRPC channel for block worker.");
```

Comment:
```suggestion
        LOGGER.warn("Failed to close gRPC channel to block worker.");
```

Reply:
Also, is there more info we can add to this message, like to which worker we were connecting to?

Reply:
Updated.